client/src/app/pages/applications/analysis-wizard/set-options.tsx

```diff
@@ -54,28 +54,29 @@ export const SetOptions: React.FC = () => {
     // Remove duplicates from array of objects based on label value (label.label)
     .filter((v, i, a) => a.findIndex((v2) => v2.label === v.label) === i);
 
-  const allTargetLabelsFromTargets = allLabelsFromTargets.filter((label) => {
-    const parsedLabel = getParsedLabel(label?.label);
-    if (parsedLabel.labelType === "target") {
-      return parsedLabel.labelValue;
-    }
-  });
+  const allTargetLabelsFromTargets = allLabelsFromTargets.filter(
+    (label) => getParsedLabel(label?.label).labelType === "target"
+  );
 
-  const allSourceLabelsFromTargets = allLabelsFromTargets.filter((label) => {
-    const parsedLabel = getParsedLabel(label?.label);
-    if (parsedLabel.labelType === "source") {
-      return parsedLabel.labelValue;
-    }
-  });
+  const allSourceLabelsFromTargets = allLabelsFromTargets.filter(
+    (label) => getParsedLabel(label?.label).labelType === "source"
+  );
 
-  const defaultTargetsAndTargetsLabels = [
-    ...defaultTargets,
-    ...allTargetLabelsFromTargets,
-  ].sort((t1, t2) => universalComparator(t1.label, t2.label));
+  const defaultTargetsAndTargetsLabels = Array.from(
+    new Map(
+      defaultTargets
+        .concat(allTargetLabelsFromTargets)
+        .map((item) => [item.label, item])
+    ).values()
+  ).sort((t1, t2) => universalComparator(t1.label, t2.label));
 
-  const defaultSourcesAndSourcesLabels = [
-    ...new Set(defaultSources.concat(allSourceLabelsFromTargets)),
-  ];
+  const defaultSourcesAndSourcesLabels = Array.from(
+    new Map(
```

Comment:
- Solution: Concatenates two arrays and processes them to ensure uniqueness based on the label property by transforming each item into key-value pairs, storing them in a new Map to overwrite duplicates, and then converting the map's values back into an array.